docs/e2e/extension-e2e-guidelines.md

```diff
@@ -314,3 +315,50 @@ Test Name: Connects to a Hardware wallet for Trezor
+
+## Page Object Model (POM)
+
+POM, or Page Object Model, is a design pattern commonly utilized for automating test cases.
+
+The Page Object is an object-oriented class that acts as an interface for the page of the application under test. Page classes contain web elements and methods to interact with these elements. While automating test cases, we create objects of these Page Classes and interact with web elements by calling the methods of these classes.
+
+### Composition of Each Class Page
+
+Each class page is composed of:
+
+- **Selectors**: HTML elements.
+- **Action Methods**: Methods for interacting with the elements.
+- **Check Methods**: Methods that assert the status of elements.
+
+The testcase spec creates the necessary page class objects and utilizes their methods to perform actions.
+
+### Best Practices
+
+- A Page Object should represent meaningful elements of a page and not necessarily a complete page. It can also be a small component, like a navbar.
+- All selectors should be kept in the page class file. Carefully define selectors for elements in page classes, opting for robust selectors since they will be extensively used across various locations.
+- The page class should contain properties and methods, or be composed of objects that expose access.
+- Page Objects should remain independent and not invoke other Page Objects to prevent circular references, ensuring they are typically isolated from each other. For handling complex workflows that require interaction across multiple pages, "Processes" should be implemented. This approach enables the incorporation of all relevant Page Objects to support specific flows, such as sending a transaction or creating a swap. A dedicated "Processes" folder is used to organize and manage these complex workflows.
+- The test cases should only call Page Object methods or Processes; they don’t interact directly with page elements.
+- Page Object methods should include detailed logs and detailed error messages in all check methods to aid in debugging tests.
+- Assertions should be included within `check_` methods. In the test specifications, `check_` methods should be called rather than making assertions directly. The reason is to make `check_` methods reusable across different test cases, with enhanced logging defined within these `check_` methods.
+- Page classes and tests should be written in TypeScript.
+- Follow the naming conventions outlined below for Page Objects, selectors, and methods.
+
+### Naming Convention
+
+#### Page Objects
+
+- Classes start with a capital letter following the word “page”, e.g., `class LoginPage`.
+- Files are named with a `-page` suffix before the `.ts` extension and use dashes for long page names, i.e., `login-page.ts`.
+
+#### Selectors
+
+- Selectors can be suffixed by the type of the element they are, like `submitButton`, `passwordInput`. Use robust selectors.
```

Comment:
Slightly better phrasing:

``` suggestion
- Selectors can be suffixed by the type of the element they represent, e.g., `submitButton`, `passwordInput`, etc. Use robust selectors.
```

Also, what does "robust" mean? Maybe this should be another bullet point, so you can expound on this a bit.

Reply:
Selector can be a no-string type as the second example explained [here](https://github.com/MetaMask/contributor-docs/blob/main/docs/e2e/extension-e2e-guidelines.md#element-locators)
but i think it makes sense to clarify here.  Also the "robust" selector mean that we should respect our selector strategy that described above.
I have added the link to selectors strategy and clarification in this commit. [9dde048](https://github.com/MetaMask/contributor-docs/pull/98/commits/9dde0481c279eca2556fdb47164d6972615764fe)

Reply:
Oh and i realized it's better to link the selectors strategy in best practises part, so i moved it there. Please check [a0a47a4](https://github.com/MetaMask/contributor-docs/pull/98/commits/a0a47a40a5b058c90948f36e0bbf7d32ed06cbdd)